File: studies/tiny/ggplot2/grouped_bar.R
library(tibble)
library(ggplot2)

df <- tribble(
  ~Contestant, ~Fruit, ~"Number Eaten",
  "Alex", "Apples", 2,
  "Alex", "Oranges", 1,
  "Alex", "Bananas", 3,
  "Jordan", "Apples", 1,
  "Jordan", "Oranges", 3,
  "Jordan", "Bananas", 2
)

ggplot(df, aes(x = factor(Fruit), y = `Number Eaten`, fill = factor(Contestant))) +
  geom_bar(stat = "identity", position = "dodge")

File: galleries/tiny/hvplot/grouped_bar.py
import hvplot.pandas
import pandas as pd

df = pd.DataFrame(
    {
        "Fruit": ["Apples", "Oranges", "Bananas", "Apples", "Oranges", "Bananas"],
        "Contestant": ["Alex", "Alex", "Alex", "Jordan", "Jordan", "Jordan"],
        "Number Eaten": [2, 1, 3, 1, 3, 2],
    }
)

df2 = df.pivot_table(index="Fruit", columns="Contestant", values="Number Eaten")
p = df2.hvplot.bar()
